tests/test_delete_stmt.py

```diff
@@ -0,0 +1,131 @@
+__author__ = 'Haleemur Ali'
+"""
+Tests to validate that the correct delete statement is
+issued for the redshift dialect of SQL.
+
+These tests use a simple transaction schema.
+
+For simple delete statments that don't have a ``WHERE`` clause
+or whose ``WHERE`` clause only refers to columns from the
+target table, the emitted query should match that emitted
+for the postgresql dialect.
+
+However, for more complex queries, an extra ``USING`` clause is required.
+
+For example, the following is valid in Postgresql:
+
+.. :code-block: sql
+
+    DELETE FROM customers
+    WHERE customers.id = orders.customer_id
+      AND orders.id < 100
+
+
+This same query needs to be written like this in Redshift:
+
+.. :code-block: sql
+    DELETE FROM customers
+    USING orders
+    WHERE customers.id = orders.customer_id
+      AND orders.id < 100
+
+"""
+
+from redshift_sqlalchemy.dialect import RedshiftDialect
+import sqlalchemy as sa
+
+
+meta = sa.MetaData()
+
+customers = sa.Table(
+    'customers', meta,
+    sa.Column('id', sa.Integer, primary_key=True, autoincrement=False),
+    sa.Column('first_name', sa.String(128)),
+    sa.Column('last_name', sa.String(128)),
+    sa.Column('email', sa.String(255))
+)
+
+orders = sa.Table(
+    'orders', meta,
+    sa.Column('id', sa.Integer, primary_key=True, autoincrement=False),
+    sa.Column('customer_id', sa.Integer),
+    sa.Column('total_invoiced', sa.Numeric(12, 4)),
+    sa.Column('discount_invoiced', sa.Numeric(12, 4)),
+    sa.Column('grandtotal_invoiced', sa.Numeric(12, 4)),
+    sa.Column('created_at', sa.DateTime),
+    sa.Column('updated_at', sa.DateTime)
+)
+
+items = sa.Table(
+    'items', meta,
```

Comment:
I don't think Redshift (or Postgresql) allows multi table deletes like MySQL does. Can't connect to an instance right now to verify this, but, I think the following would fail on Redshift

    DELETE FROM table1, table2

Also, self-joins in `DELETE` are not supported by the extension I wrote, but a self-join is a more difficult construct, and I feel it is reasonable to expect users to produce hand-written queries for this purpose.

Statements like below are fully supported and are also tested (`test_delete_stmt_joinedwhereclause2`)

    DELETE FROM table1 
    USING table2, table3 
    WHERE table2.col1  = table1.col1
    AND table3.col2 = table1.col2

I'll add tests to cover these cases:

    DELETE FROM table1 
    WHERE table1.col1 IN (SELECT col2 FROM table2)

and

    DELETE FROM table1
    USING table2
    WHERE table1.col1 IN (SELECT col1 FROM table3)
    AND table1.col2 = table2.col2